src/game/handlers/upStat.js

```diff
@@ -11,19 +12,22 @@ import { level } from '../core/level'
 import { reject } from './errors'
 
 import { capitalize } from './helpers'
-import { getStatCost, getCurrentStatPoints, statIds } from './statHelpers'
+import { statUpgradeCost, unspentStatPoints, statIds } from './statHelpers'
 
-function increaseStat (dao, char, statId) {
-  const amount = char[statId]
+function updateStat (dao, char, statId, changeAmount = 1) {
+  const currentStatScore = char[statId]
   const query = pick(['_id', 'updatedAt'], char)
+
   return dao.character.update(query, {
-    $set: { [statId]: amount + 1 },
+    $set: { [statId]: currentStatScore + changeAmount },
   })
 }
 
 export default function call (dao, provider, _, msg) {
-  const statName = msg.matches[1]
+  const statName = msg.matches[2]
   const statId = statIds[statName]
+  const changeType = msg.matches[1] === 'refund' ? 'refund' : 'upgrade'
```

Comment:
The only way to call that handler right now is "refund", "up" and "upgrade" because of the regex in `route.js`. That line you reviewed is mostly to streamline "up" as "upgrade" so I don't have to be checking "up or upgrade" everywhere else.

I guess this could be better for handling future modifications that what it is in the current PR

    const changeType = msg.matches[1] === 'up' ? 'upgrade' : msg.matches[1]

do you like that a bit more? or did you have anything else in mind?